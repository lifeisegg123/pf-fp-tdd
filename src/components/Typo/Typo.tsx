import { ReactNode } from "react";
import { ComponentBaseProps } from "src/interfaces";

interface TypoProps extends ComponentBaseProps {
  children: ReactNode;
  as?: "p" | "h1" | "h2" | "h3" | "h4" | "h5" | "h6" | "span" | "label";
}

export function Typo({ as = "p", ...rest }: TypoProps) {
  const Root = as;
  return <Root {...rest} />;
}
