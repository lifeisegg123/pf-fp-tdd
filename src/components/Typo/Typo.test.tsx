import { Typo } from "./Typo";
import { render, fireEvent } from "@testing-library/react";

describe("Component/Typo", () => {
  it("should match snapshot", () => {
    const { container } = render(<Typo>test</Typo>);
    expect(container).toMatchSnapshot();
  });

  it("should render text", () => {
    const { getByText } = render(<Typo>test</Typo>);

    expect(getByText("test")).toBeInTheDocument();
  });
});
