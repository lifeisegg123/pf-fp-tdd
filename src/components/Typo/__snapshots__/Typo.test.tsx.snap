// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component/Typo should match snapshot 1`] = `
<div>
  <p>
    test
  </p>
</div>
`;
